frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+  RequestSizeInput,
+  withHandlePromise,
+  humanizeBinaryBytesWithoutB,
+} from '@console/internal/components/utils/index';
+import { StorageClassDropdown } from '@console/internal/components/utils/storage-class-dropdown';
+import {
+  createModalLauncher,
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+} from '@console/internal/components/factory';
+import './_add-capacity-modal.scss';
+import { k8sPatch, K8sResourceKind } from '@console/internal/module/k8s';
+import { OCSServiceModel } from '../../../models';
+
+export const AddCapacityModal = withHandlePromise((props: AddCapacityModalProps) => {
+  const { obj, close, cancel } = props;
+
+  const [requestSizeUnit] = 'Ti';
+  const [requestSizeValue, setRequestSizeValue] = React.useState('');
+  const [storageClass, setStorageClass] = React.useState('');
+  const [inProgress, setProgress] = React.useState(false);
+  const [errorMessage, setError] = React.useState('');
+
+  const getPresentValue = () => {
+    const currentValue =
+      obj.spec.storageDeviceSets[0].dataPVCTemplate.spec.resources.requests.storage;
+    const unit = currentValue.replace(/[0-9]/g, '');
+    const val = parseInt(currentValue, 10);
+    const humanizedSize = humanizeBinaryBytesWithoutB(val, unit, requestSizeUnit);
+    return humanizedSize;
+  };
+  getPresentValue();
```

Comment:
Why this function is called twice? Its being called inside submit function as well.

Reply:
ack, Its not required outside